src/components/views/dialogs/BaseDialog.js

```diff
@@ -45,19 +43,35 @@ export default React.createClass({
 
         // children should be the content of the dialog
         children: React.PropTypes.node,
+
+        // Id of content element
+        // If provided, this is used to add a aria-describedby attribute
+        contentId: React.PropTypes.string,
+    },
+
+    componentDidMount: function() {
+        // Retrieve the root node of the Riot application outside the dialog
+        this.applicationNode = document.getElementById('matrixchat');
```

Comment:
Originally I wanted it to have it the way you are suggesting. I have started by using FocusTrap component on the dialog parent div element. I have found out FocusTrap is not working well when placed outside the dialog (i.e. managed via the Modal manager) so I moved it into the dialog. I think hiding the main application node complements trapping the keyboard focus it's why I have added it into the dialog not to its parent. Later on I have discovered we are sometimes loading other elements e.g. spinner into the modal. I was unable to assure if all the components loaded into the modal should trap the keyboard focus and disable screen reader access to the main application so this is only supporting my doupt. If you are sure we are only loading keyboard navigable stuff into the modal then I'm happy to move the aria-hide stuff into the Modal but I won't be moving FocusTrap thingy as it is not working very well outside the dialog. If there is a chance that we might occassionally load something into the modal that can't be navigated by the keyboard alone then implementing aria-hidden at this level would cause screen reader users to be locked in the modal without possible exit path out of it.

Reply:
>If there is a chance that we might occassionally load something into the modal that can't be navigated by the keyboard

I'd say that would be a bug anyway.

If we don't check the number of dialogs open before deciding whether to apply `aria-hidden` to the root element, we might end up with races and *sometimes* the `aria-hidden` property will be set as expected.